tests/unit/test_graph_parser.py

```diff
@@ -139,6 +139,36 @@ def test_graph_syntax_errors_2(seq, graph, expected_err):
             'Invalid cycle point offsets only on right',
             id='no-cycle-point-RHS'
         ),
+        # See https://github.com/cylc/cylc-flow/issues/6523
+        # For the next 4 tests:
+        param(
+            # Yes I know it's circular, but it's here to
+            # demonstrate that the test below is broken:
+            "foo:finished => foo",
+            'Output foo:succeeded can\'t be both required and optional',
+            id='finish-implies-success-optional'
+        ),
+        param(
+            "foo[-P1]:finish => foo",
+            'Output foo:succeeded can\'t be both required and optional',
+            id='finish-implies-success-optional-offset'
+        ),
+        param(
+            "foo[-P1]:succeeded | foo[-P1]:failed => bar",
+            # order of outputs varies in the error message
+            'must both be optional if both are used',
+            id='succeed-or-failed-mustbe-optional'
+        ),
+        param(
+            "foo[-P1]:succeeded? | foo[-P1]:failed? => foo",
+            # Order of processing can result in different error messages
```

Comment:
I fixed this in my version of the pr https://github.com/hjoliver/cylc-flow/pull/62

Reply:
took it, thanks